pkg/problemclient/problem_client.go

```diff
@@ -49,12 +50,17 @@ type nodeProblemClient struct {
 }
 
 // NewClientOrDie creates a new problem client, panics if error occurs.
-func NewClientOrDie() Client {
+func NewClientOrDie(apiServer string) Client {
 	c := &nodeProblemClient{clock: util.RealClock{}}
-	cfg, err := restclient.InClusterConfig()
+
+	// error is ignored because we have checked it after command line is parsed.:)
+	uri, _ := url.Parse(apiServer)
```

Comment:
@Random-Liu Just as the comment said, the error is ignored. Because the `apiServer` has been validated after command line argument is parsed in `node_problem_detector.go`'s `validateCmdParams` function. It will fatal in there if the format of `apiServer` URI is invalid.

This will it fail faster if some precondition is not met.